cmd/kubeadm/app/master/selfhosted.go

```diff
@@ -333,3 +361,142 @@ func getSchedulerDeployment(cfg *kubeadmapi.MasterConfiguration, volumes []v1.Vo
 func buildStaticManifestFilepath(name string) string {
 	return path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, "manifests", name+".yaml")
 }
+
+func filesToSecret(files []string, client *clientset.Clientset) error {
+	dataMap := make(map[string][]byte, 0)
+
+	for _, file := range files {
+		name := path.Base(file)
+		data, err := ioutil.ReadFile(file)
+		if err != nil {
+			return err
+		}
+		dataMap[name] = data
+
+		if err = os.Remove(file); err != nil {
+			return err
+		}
+	}
+
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "kubeadm",
+			Namespace: metav1.NamespaceSystem,
+		},
+		Type: v1.SecretTypeOpaque,
+		Data: dataMap,
+	}
+
+	if _, err := client.Secrets(metav1.NamespaceSystem).Create(secret); err != nil {
+		return err
+	}
+	return nil
+}
+
+func apiServerSecretVolume() v1.Volume {
+	return v1.Volume{
+		Name: "k8s",
+		VolumeSource: v1.VolumeSource{
+			Secret: &v1.SecretVolumeSource{
+				SecretName: "kubeadm",
+				Items: []v1.KeyToPath{
+					v1.KeyToPath{
+						Key:  "apiserver-kubelet-client.crt",
+						Path: "pki/apiserver-kubelet-client.crt",
```

Comment:
These are hard coded for now, until we decide if this is the direction we want to go. I can clean up with another push.